src/gui/src/components/explorer-grid/selected-item/tabs-insight.tsx

```diff
@@ -0,0 +1,203 @@
+  selector: string
+  category: SocketSecurityDetails['category']
+  description: string
+}) => {
+  return (
+    <div className="duration-250 grid cursor-default grid-cols-10 gap-2 rounded-sm px-2 py-4 transition-all hover:bg-muted">
+      <div className="col-span-3 flex w-full items-start">
+        <InsightBadge
+          color={getAlertColor(severity)}
+          tooltipContent={severity}>
+          {selector}
+        </InsightBadge>
+      </div>
+      <div className="col-span-2 flex w-full">
+        <p className="text-sm">{category}</p>
+      </div>
+      <div className="col-span-5 flex">
+        <p className="text-pretty text-sm">{description}</p>
+      </div>
+    </div>
+  )
+}
+
+export const InsightTabContent = () => {
+  const [insightsExpanded, setInsightsExpanded] =
+    useState<boolean>(false)
+  const { selectedItem, securityArchive } = useSelectedItem()
+  const [allPotentialInsights, setAllPotentialInsights] = useState<
+    SocketSecurityDetails[]
+  >(() => Object.values(SOCKET_SECURITY_DETAILS))
+  const [securityAlerts, setSecurityAlerts] = useState<
+    SocketSecurityDetails[] | undefined
+  >(() => getSecurityAlerts(selectedItem, securityArchive))
+
+  const toggleInsights = () => {
+    setInsightsExpanded(prev => !prev)
+  }
+
+  return (
+    <TabsContent value="insights">
+      <section className="mt-2 flex flex-col px-6 py-4">
+        {securityAlerts && securityAlerts.length > 0 && (
+          <>
+            <InsightHeader
+              items={securityAlerts}
+              setItems={setSecurityAlerts}
+            />
+
+            <div className="flex flex-col divide-y-[1px] divide-muted">
+              {securityAlerts.map((alert, idx) => (
+                <InsightItem
+                  key={idx}
+                  selector={alert.selector}
+                  severity={alert.severity}
+                  description={alert.description}
+                  category={alert.category}
+                />
+              ))}
+            </div>
+          </>
+        )}
+      </section>
+
+      <AnimatePresence>
+        {insightsExpanded && (
+          <motion.section
+            initial={{ opacity: 0 }}
+            animate={{ opacity: 1 }}
+            exit={{ opacity: 0 }}
+            className="border-t-[2px] border-muted px-6 py-4">
+            <div className="mt-6 flex flex-col">
+              <p className="mb-6 w-full text-base font-medium">
+                All Potential Insights
+              </p>
+              <InsightHeader
+                items={allPotentialInsights}
+                setItems={setAllPotentialInsights}
+              />
+
+              <div className="flex flex-col divide-y-[1px] divide-muted">
+                {allPotentialInsights.map((insight, idx) => (
+                  <InsightItem key={idx} {...insight} />
+                ))}
+              </div>
+            </div>
+          </motion.section>
+        )}
+      </AnimatePresence>
```

Comment:
I'm seeing duplicated entries when testing locally, might be worth double-checking on your end:

<img width="721" alt="Screenshot 2025-03-31 at 9 27 49 PM" src="https://github.com/user-attachments/assets/9ff98b35-1a90-45ce-bdc6-a0658f9fb8ce" />